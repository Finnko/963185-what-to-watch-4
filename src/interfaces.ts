interface Movie {
  id: number,
  title: string
  poster: string,
  thumb: string,
  preview: string,
  videoSrc: string,
  releaseYear: number,
  description: string,
  backgroundColor: string,
  backgroundImage: string,
  director: string,
  rating: number,
  ratingCount: number,
  runTime: number,
  actors: string[],
  genre: string,
  isFavorite: boolean,
}

interface Review {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: {
    id: number,
    name: string,
  }
}

interface User {
  id: number,
  email: string,
  name: string,
  avatarUrl: string,
}

interface FormField {
  value: string,
  valid: boolean,
  touched: boolean,
  validation: {
    required: boolean,
    email?: boolean ,
    minLength?: number,
  }
}

export {
  Movie,
  Review,
  User,
  FormField,
}
