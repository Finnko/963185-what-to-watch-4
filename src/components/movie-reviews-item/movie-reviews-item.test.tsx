import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MovieReviewsItem from './movie-reviews-item';
import {Review} from "../../interfaces";

const mockReview: Review = {
  comment: `Good comment!`,
  date: `2020-07-09T16:06:01.831Z`,
  id: 2,
  rating: 9.3,
  user: {
    id: 15,
    name: `Ozzy Osbourne`
  },
};

describe(`MovieReviewsItem component render correctly`, () => {
  it(`Should MovieReviewsItem component render correctly`, () => {
    const tree = renderer
      .create(
          <MovieReviewsItem
            review={mockReview}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
