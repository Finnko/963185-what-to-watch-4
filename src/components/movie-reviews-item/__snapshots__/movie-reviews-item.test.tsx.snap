// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReviewsItem component render correctly Should MovieReviewsItem component render correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      Good comment!
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Ozzy Osbourne
      </cite>
      <time
        className="review__date"
        dateTime="2020-07-09"
      >
        July 09, 2020
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    9.3
  </div>
</div>
`;
